src/Compilers/CSharp/Test/Symbol/Symbols/Source/NullablePublicAPITests.cs

```diff
@@ -2217,5 +2217,35 @@ public void M(string s)
             Assert.Equal("s", symbol.Name);
             Assert.Equal(SpecialType.System_String, symbol.Type.SpecialType);
         }
+
+        [Fact, WorkItem(37879, "https://github.com/dotnet/roslyn/issues/37879")]
+        public void MissingSymbols_ReinferedParent()
+        {
+            var source = @"
+class C
+{
+    public void A<T>(T t) where T:class
+    {
+        var c = new F<T>[] { }.Select(v => new { Value = v.Item }).ToArray();
+    }
+    private class F<T>
+    {
+        public F(T oldItem) => Item = oldItem;
+        public T Item { get; }
+    }
+}";
+
+            var comp = CreateCompilation(source, options: WithNonNullTypesTrue());
+
+            var syntaxTree = comp.SyntaxTrees[0];
+            var root = syntaxTree.GetRoot();
+            var model = comp.GetSemanticModel(syntaxTree);
+
+            var select = root.DescendantNodes().OfType<IdentifierNameSyntax>().Where(i => i.Identifier.ValueText == "Select").Single();
+            var symbolInfo = model.GetSymbolInfo(select);
+
+            Assert.Null(symbolInfo.Symbol);
```

Comment:
>Assert.Null(symbolInfo.Symbol); [](start = 12, length = 31)

What did we get before the change?

Reply:
We returned a bad symbol that had a `null` `ContainingType`, which the IDE was throwing on.

---
In reply to: [314005550](https://github.com/dotnet/roslyn/pull/37964#discussion_r314005550) [](ancestors = 314005550)

Reply:
> We returned a bad symbol that had a null ContainingType, which the IDE was throwing on.

What was that symbol and where it was coming from? Was it an error MethodSymbol from the bound node?

---
In reply to: [314007924](https://github.com/dotnet/roslyn/pull/37964#discussion_r314007924) [](ancestors = 314007924,314005550)

Reply:
Yes, exactly.

---
In reply to: [314009018](https://github.com/dotnet/roslyn/pull/37964#discussion_r314009018) [](ancestors = 314009018,314007924,314005550)